scrapy/pipelines/files.py

```diff
@@ -248,6 +250,42 @@ def persist_file(self, path, buf, info, meta=None, headers=None):
         )
 
 
+class FTPFilesStore(object):
+    
+    def __init__(self, uri):
+        assert uri.startswith('ftp://')
+        u = urlparse(uri)
+        self.ftp = FTP()
+        self.ftp.connect(u.hostname, u.port or '21')
+        self.ftp.login(u.username, u.password)
```

Comment:
Maybe we could use the existing `FTP_USERNAME` and `FTP_PASSWORD` settings as fallbacks if the URL does not define a username or password?

See how it’s done for `AWS_ACCESS_KEY_ID` and similar in `FilesPipeline`.

Reply:
Cool, will add that :)